lib/Doctrine/Common/Persistence/Mapping/Driver/SymfonyFileLocator.php

```diff
@@ -172,9 +186,17 @@ public function getAllClassNames($globalBasename = null)
 
                     // NOTE: All files found here means classes are not transient!
                     if (isset($this->prefixes[$path])) {
-                        $classes[] = $this->prefixes[$path].'\\'.str_replace('.', '\\', $fileName);
+
+                        // Calculate namespace suffix for given prefix as a relative path from basepath to file path
+                        $nsSuffix = strtr(
+                            substr(realpath($file->getPath()), strlen(realpath($path))),
+                            $this->nsSeparator,
+                            '\\'
+                        );
+
+                        $classes[] = $this->prefixes[$path] . $nsSuffix . '\\' .str_replace($this->nsSeparator, '\\', $fileName);
```

Comment:
There is an issue here when Doctrine `auto_mapping` is enabled in Symfony and having a subdirectory like `Resources/config/doctrine/model` for instance.